### PR TITLE
third pass at Demon Hunter

### PR DESCRIPTION
**Pull request by ToppleTheNun**
third pass at Demon Hunter (#5087)

add Darkglare Boon support

add support for Soulrend

add very early support for Any Means Necessary

update The Hunt spell IDs

add support for Inner Demon talent

update Eye Beam base CD

add support for Ragefire talent

rename Master of the Glaives to Master of the Glaive

correct Fel Rush CD calculation

correct Infernal Strike CD calculation

regenerate talents

blur baseline for havoc, ca in class tree

Sigil of Flame is a class talent

improve CDR calculation for Infernal Strike

fix ImmolationAura for havoc by making it shared

update metamorphosis cooldown calculation

encapsulate Infernal Strike and Fel Rush CD calculation

add Sigil of Chains versions for Precise and Concentrated

correct location of Furious Gaze statistic

add support for Growing Inferno

remove old Void Reaver support

re-add support for Burning Hatred

correct duration of Demonic proc

mark Havoc as ready for use

tested with WCL reports:
/report/NM1kZRCv6nYhWft4/5-Heroic+Eranog+-+Kill+(6:18)/Saeil/standard/overview
/report/hPVLCkGMmr31t9jw/15-Heroic+Eranog+-+Kill+(8:05)/Jedith/standard/overview

### CODE REVIEW

File: src/analysis/retail/demonhunter/havoc/modules/spells/Blur.tsx [hunk 1]
```diff
@@ -1,5 +1,4 @@
-import Analyzer, { Options } from 'parser/core/Analyzer';
-import { TALENTS_DEMON_HUNTER } from 'common/TALENTS';
+import Analyzer from 'parser/core/Analyzer';
```
Comment (emallson) [inline src/analysis/retail/demonhunter/havoc/modules/spells/Blur.tsx:1]: rip vengeance getting a good defensive cd :(

File: src/analysis/retail/demonhunter/vengeance/modules/Abilities.tsx [hunk 2]
```diff
@@ -259,7 +270,19 @@ class Abilities extends CoreAbilities {
...
+            ABYSSAL_HASTE_SCALING[
+              combatant.getTalentRank(TALENTS_DEMON_HUNTER.ABYSSAL_HASTE_VENGEANCE_TALENT)
+            ];
+          const erraticFelheartReduction =
+            ERRATIC_FELHEART_SCALING[
+              combatant.getTalentRank(TALENTS_DEMON_HUNTER.ERRATIC_FELHEART_TALENT)
+            ];
+          const flatReduced = baseCooldown - abyssalHasteReduction;
+          return flatReduced - flatReduced * erraticFelheartReduction;
+        },
```
Comment (emallson) [inline src/analysis/retail/demonhunter/vengeance/modules/Abilities.tsx]: this is complex enough that you might want to move it to be an exported function in the `InfernalStrike` file.

**Review by emallson — COMMENTED** [review]
just a minor comment. stuff looks good. i know this is marked as a draft but i had time to review so i did.

if you are happy with what is here and want to merge, we can get it in.

File: src/analysis/retail/demonhunter/havoc/modules/spells/Blur.tsx (hunk 1 above)
Comment (ToppleTheNun) [reply src/analysis/retail/demonhunter/havoc/modules/spells/Blur.tsx:1]: 😢

**Comment by ToppleTheNun** [comment]
> marked as a draft

It's mostly marked as draft until I can get enough logs to confirm that things are behaving as intended. I'm hoping over the weekend more people will feed me logs so I can verify both Havoc and Vengeance.

**Comment by ToppleTheNun** [comment]
This is ready for review at this point.

**Review by emallson — APPROVED** [review]

**Comment by emallson** [comment]
Only not that I have is that marking havoc as full support so early in the cycle may be premature but also it removes the "Are you sure?" screen so i think its okay. (I think sref might have done the same thing for rdru?)

**Pull request merged**